### ISSUE TITLE
Refactor for server side usage

### ISSUE THREAD

**Issue by bkniffler** [mikemintz/rethinkdb-websocket-client#1]
Currently its not possible to use this library on the server side (e.g "self not defined"). Especially with react-rethinkdb, it would be very useful to be able to use it in an isomorphic environment.

For example: https://github.com/iam4x/isomorphic-flux-boilerplate

**Comment by mikemintz** [comment]
Implemented in https://github.com/mikemintz/rethinkdb-websocket-client/pull/2

I'll merge this into master and bump the version once I confirm it works with react-rethinkdb

**Comment by mikemintz** [comment]
Merged into master and bumped version to 0.2.0. If you're loading this directly, as opposed to via react-rethinkdb, you'll need to use this require path for server-side:

``` js
var RethinkdbWebsocketClient = require('rethinkdb-websocket-client/dist/node');
```

For react-rethinkdb server side support, I'll update that soon.

**Comment by bkniffler** [comment]
Great, thanks! 
For server-side support (talking about server-side rendering for no-js or initial page load especially), will the queries run through the query whitelists and support all the auth stuff?

**Comment by mikemintz** [comment]
Good question. Yes, the queries will run through the same query whitelist infrastructure. The HTTP server connects to the same websocket server that the client does, as opposed to connecting directly to rethinkdb.

In [`examples/isomorphic`](https://github.com/mikemintz/react-rethinkdb/tree/master/examples/isomorphic), I had set `unsafelyAllowAnyQuery: true` but that was just for simplicity.

The part that takes some thinking is sending the auth token from the client to the HTTP server. In [`examples/chat`](https://github.com/mikemintz/react-rethinkdb/tree/master/examples/chat), the auth token is stored in localStorage, so the HTTP server wouldn't be able to replicate the client's websocket connection. But I imagine if we stored it in a cookie instead, it would be possible. I don't know what the best practices are for authenticated requests in isomorphic apps.
